airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/task/ProcessRecordsTask.kt

```diff
@@ -36,29 +36,37 @@ class DefaultProcessRecordsTask(
     private val deserializer: Deserializer<DestinationMessage>,
 ) : ProcessRecordsTask {
     override suspend fun execute() {
+        val log = KotlinLogging.logger {}
+
+        log.info { "Processing records from ${fileEnvelope.batch.localFile}" }
         val nextBatch =
             withContext(Dispatchers.IO) {
-                val records =
-                    fileEnvelope.batch.localPath
-                        .bufferedReader(Charsets.UTF_8)
-                        .lineSequence()
-                        .map {
-                            when (val message = deserializer.deserialize(it)) {
-                                is DestinationStreamAffinedMessage -> message
-                                else ->
-                                    throw IllegalStateException(
-                                        "Expected record message, got ${message::class}"
-                                    )
+                fileEnvelope.batch.localFile.toFileReader().use { reader ->
+                    val records =
+                        reader
+                            .lines()
+                            .map {
+                                when (val message = deserializer.deserialize(it)) {
+                                    is DestinationStreamAffinedMessage -> message
+                                    else ->
+                                        throw IllegalStateException(
+                                            "Expected record message, got ${message::class}"
+                                        )
+                                }
+                            }
+                            .takeWhile {
+                                it !is DestinationStreamComplete &&
+                                    it !is DestinationStreamIncomplete
                             }
-                        }
-                        .takeWhile {
-                            it !is DestinationStreamComplete && it !is DestinationStreamIncomplete
-                        }
-                        .map { it as DestinationRecord }
-                        .iterator()
-                streamLoader.processRecords(records, fileEnvelope.batch.totalSizeBytes)
+                            .map { it as DestinationRecord }
+                            .iterator()
+                    streamLoader.processRecords(records, fileEnvelope.batch.totalSizeBytes)
+                }
             }
 
+        log.info { "Processing complete, deleting ${fileEnvelope.batch.localFile}" }
+        fileEnvelope.batch.localFile.delete()
```

Comment:
does it need to be in a try/finally block?

Reply:
Yeah. Right now if this fails, it will fail the job, but that won't necessarily always be true.

I'll explore making the file itself closable, otherwise try/finally

Reply:
Actually no obviously that's dumb. Close != delete. I'll just try/finally.